secure-device-storage/sds-android/securedevicestorage/src/main/java/de/adorsys/android/securedevicestorage/SecurePreferences.java

```diff
@@ -87,7 +129,26 @@ public static void clearAllValues(@NonNull Context context) {
         clearAllSecureValues(context);
     }
 
-    @SuppressLint("CommitPrefEdits")
+    public static boolean compareHashedCredential(@NonNull String currentCredential,
+                                                  @NonNull String keyOfSecureCredential,
+                                                  @NonNull Context context) {
+
+        String securedCredential = getValue(keyOfSecureCredential, context, SecureMethod.METHOD_HASH);
+        String salt = getValue(SALT_PREFIX + keyOfSecureCredential, context, SecureMethod.METHOD_HASH);
+        String hashedCurrentCredential = KeystoreTool.getSHA512(currentCredential, salt);
+
+        if (hashedCurrentCredential != null) {
+            return hashedCurrentCredential.equals(securedCredential);
+        } else {
+            if (BuildConfig.DEBUG) {
+                Log.e(SecurePreferences.class.getName(),
```

Comment:
I think it's a good idea to throw a custom RuntimeException here so that the developer can get feedback that it didn't work